### ISSUE TITLE
Audio from sound files starts playing a couple of seconds out in the file

### ISSUE THREAD

**Issue by albertony** [mltframework/shotcut#571]
When I add add audio clips (to source player and/or timeline) and starts playing them, the playback in Shotcut starts 1-3 seconds out in the file. When exporting my Shotcut project the same issue is present in the exported file. For short sound-effects it means I don't hear the sound at all, or just some short part of its end. The same files plays correct in VLC and Audacity.

I first noticed this with a bunch of MP4 files I created, containing 320 kbps AAC audio. I measured the issue in one of these, comparing playback in Shotcut with Audacity, and Shotcut starts the playback 3.2 seconds out in the clip. If I export it to wav or flac from Audacity and add that to Shotcut the issue is reduced to 1 second. I worked around it in my latest (and first ever) project by editing the files in Audacity, adding 1 second silence in the beginning and then saving as flac files that I used in Shotcut.

Not sure if the same thing happens for all files, maybe not, but I do get the same issue when I try some old wav files I had on my computer (From Homer Simpson's Wohoo.wav I either don't hear anything or just a short noice from the end).

In case you want a file the issue is experienced with, I just tried the "Car start garage" sound from freesoundeffects.com (https://www.freesoundeffects.com/free-track/carstartgarage-466329/Car%20start%20garage/), and that has the same issue: Both the mp3 and wav files starts a second or so too late when played back in Shotcut.

Tested on Windows 10 (1709) with Shotcut version 18.05.08.

**Comment by ddennedy** [comment]
I reproduce this using the supplied sound effect in the Source player when Settings > Realtime is enabled. That needs work agreed. However, when I add the sound clip to the timeline of an existing project I do not reproduce it, and I do not reproduce it in the exported timeline. Also, if I open the mp3 in Shotcut, and export it was WAV (not in timeline), the result plays as expected outside Shotcut.

Shotcut v18.05 introduced support for mono and surround sound, which might have had an impact (it has affected scrubbbing, for example). Please test v18.03 since I am having trouble to reproduce it as you described.

**Comment by albertony** [comment]
Ok, so there is more to it (as is often the case)..

I can confirm that (in v18.05) turning off Settings > Realtime avoids the issue with the wav/mp3 sample file in the source player, and that (with or without Realtime) the timeline/export does not have the issue in this test case.

But the mp4 audio files that I originally wanted to use do have the problem - in source player, timeline and export, and with or without realtime setting. I can try them with v18.03, probably tomorrow.

It could be that after struggling for a while to get the first 3 seconds of my mp4 files to be played, I got put off by the fact that converting them to wav/flac still seemed to have the issue - but then I might have just tested them in the source player and without considering the Realtime setting.

**Comment by ddennedy** [comment]
> the mp4 audio files that I originally wanted to use do have the problem

Please share one or describe how you created it in case I have the tool.

**Comment by albertony** [comment]
Here you go (had trouble uploading it as mp4 or zip, but when I used 7-zip to pack it as gz it seems to have worked..)

[Record Player Scratching.gz](https://github.com/mltframework/shotcut/files/2036888/Record.Player.Scratching.gz)

**Comment by albertony** [comment]
Glad to see you have done some improvements. It is related to the source player issue and not the mp4 issue if I understand correctly. I just tried v18.03 with my problematic mp4 files, and experience the same issue in that version.
PS: I really enjoyed working with Shotcut, really impressed of its powers! I experienced a couple of annoyances that I plan to report as issues, but hope you considers it as a contribution and not as complaining.
